src/transformers/trainer.py

```diff
@@ -815,10 +815,12 @@ def create_optimizer(self):
             else:
                 self.optimizer = optimizer_cls(optimizer_grouped_parameters, **optimizer_kwargs)
 
-        if is_sagemaker_mp_enabled():
-            self.optimizer = smp.DistributedOptimizer(self.optimizer)
+            if is_sagemaker_mp_enabled():
+                self.optimizer = smp.DistributedOptimizer(self.optimizer)
+
+        return self.optimizer
 
-    def create_scheduler(self, num_training_steps: int):
+    def create_scheduler(self, optimizer: torch.optim.Optimizer, num_training_steps: int):
```

Comment:
hmm, that would break back-compat as the new arg is added before the existing one - and if someone used a positional arg to pass `num_training_steps` it now will break.

Reply:
may be putting it last and making it optional? So by default it works as before but only if the optional `optimizer is not None` than we use that?

Reply:
Good point.